src/main/java/io/vertx/ext/mail/mailencoder/Utils.java

```diff
@@ -203,7 +214,11 @@ static String encodeEmailList(List<String> addresses, int index) {
       if (!name.isEmpty()) {
         if (mustEncode(name)) {
           boolean hadSpace = false;
-          if (index + 12 >= 71) {
+          // 12 is the size of the string " (=?UTF-8?Q?".
+          // 71 is 76 - 5, and 5 is the size of "/n " plus the minimal size of the encode char (3 bytes)
+          // However, the encoded char may be larger, for example "=AA=BB=CC",
+          // and needs to fold to the next line as a whole, and in this case it will leave a " (=?UTF-8?Q??="
+          if (encodeHeader(name, index + 2).contains("=?UTF-8?Q??=") || (index + 12 >= 71)) {
```

Comment:
I don't understand this line about the `encodeHeader(name, index + 2)`, and the result of `encodeHeader()` should not contain `=?UTF-8?Q??=` IMO because email address should not contain line break.

As what I understand it, we are trying to check if the index plus the encoded name length is larger than line limit (`76`), and the encoded name at least contains `=?UTF-8?Q?=XX` which is `13`, not `12`, the 5 left is because we will include space ` `, `(`, and `)` for the email name, and the `?=` at the end of encoding. So maybe just change it to `if (index + 13 >= 71) {` should work.

Reply:
I updated the code to
```java
if ((index + 17 >= 76) || encodeHeader(name, index + 2).contains("=?UTF-8?Q??=")) {}
```
At this point, the email has already been appended, it's time to append the name, and the name contains at least a character that must be encoded. 
First, let's look at `index + 17 >= 76`. Here is my understanding. Suppose the first character of the name must be encoded, and the value is "=AA", we need to append `" (=?UTF-8?Q?=AA"` here, which is 15 bytes. I'm not quite sure why it's 17 in the original code, maybe it means an extra of `"?="`, or `")\n"` or `"),"`.  The first character in the name may not need to be encoded, but we just consider the worst case here. 
However, it's not enough because if the encoded result of the first character is, for example, `"=AA=BB=CC"` they need to be folded into the next line as a whole. In that case, we may end up with a `"=?UTF-8?Q??="` which is not allowed according to your reply. That's why we have `(encodeHeader(name, index + 2).contains("=?UTF-8?Q??=")`, `+ 2` is due to `" ("`.

Reply:
I see why we get `=?UTF-8?Q??=` now, thank you. But I don't like the idea to check the length by encoding it first, since we can check it by counting the known character numbers.

May I know that is there any case that I may miss if we check the length by  `if (index + 13 >= 71)` ?  thanks :)

Reply:
Hello @gaol 
I updated the code again and hopefully is clear and obvious this time.
The logic can be expressed that the remaining space must be able to hold the encoded first character of the name.
I also add a unit test for this.

by the way, `if (index +13 >71)  {` doesn't work because it doesn't take Unicode char into account. `=XX` might need to be folded into the next line and live a `"=?UTF-8?Q??="`. (Yes, many things twisted together)